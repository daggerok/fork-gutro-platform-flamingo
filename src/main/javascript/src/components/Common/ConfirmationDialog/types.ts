export type ConfirmationDialogProps = {
    onOkClick: (event: any) => void;
    onCancelClick: (event: any) => void;
    title: string;
    subTitle: string | undefined;
}