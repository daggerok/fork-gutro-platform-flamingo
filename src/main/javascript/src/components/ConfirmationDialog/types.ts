export type ConfirmationDialogProps = {
    onOkClick: (event: React.MouseEvent) => void;
    onCancelClick: (event: React.MouseEvent) => void;
    title: string;
    subTitle: string | undefined;
}