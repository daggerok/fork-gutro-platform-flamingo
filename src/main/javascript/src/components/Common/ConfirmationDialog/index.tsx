import React from 'react';
import { Modal, Result, Button } from 'antd';

import { ConfirmationDialogProps } from './types';

const ConfirmationDialog: React.FC<ConfirmationDialogProps> = ({
  onOkClick,
  onCancelClick,
  title,
  subTitle,
}: ConfirmationDialogProps) => {
  return (
    <Modal
      visible={true}
      closable={false}
      footer={null}
      data-testid="confirmation-dialog"
    >
      <Result
        status="warning"
        title={title}
        subTitle={subTitle}
        extra={[
          <Button
            type="default"
            key="cancel"
            onClick={onCancelClick}
            data-testid="confirmation-cancel-button"
          >
            Cancel
          </Button>,
          <Button
            type="primary"
            key="ok"
            onClick={onOkClick}
            data-testid="confirmation-ok-button"
          >
            OK
          </Button>,
        ]}
      />
    </Modal>
  );
};

export default ConfirmationDialog;
