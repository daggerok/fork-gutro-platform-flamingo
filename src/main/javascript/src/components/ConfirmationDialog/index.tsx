import React from 'react';
import { Modal, Result, Button } from 'antd';

type ConfirmationDialogProps = {
  onOkClick: (event: React.MouseEvent) => void;
  onCancelClick: (event: React.MouseEvent) => void;
  title: string;
  subTitle: string | undefined;
}

const ConfirmationDialog: React.FC<ConfirmationDialogProps> = ({
  onOkClick,
  onCancelClick,
  title,
  subTitle,
}: ConfirmationDialogProps) => {
  return (
    <Modal
      visible={true}
      closable={false}
      footer={null}
    >
      <Result
        status="warning"
        title={title}
        subTitle={subTitle}
        extra={[
          <Button
            danger
            type="primary"
            key="ok"
            onClick={onOkClick}
          >
            OK
          </Button>,
          <Button
            type="default"
            key="cancel"
            onClick={onCancelClick}
          >
            Cancel
          </Button>,
        ]}
      />
    </Modal>
  );
};

export default ConfirmationDialog;
