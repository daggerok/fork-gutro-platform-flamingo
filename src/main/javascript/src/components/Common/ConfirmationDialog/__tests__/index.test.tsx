import React from 'react';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event'
import ConfirmationDialog  from '..';

let onOkClickSpy = jest.fn();
let onCancelClickSpy = jest.fn();

let component: any = null;

const BASE_PROPS = {
  onOkClick: onOkClickSpy,
  onCancelClick: onCancelClickSpy,
  title: 'A nice title',
  subTitle: 'A nicer subtitle',
}

describe('<ConfirmationDialog />', () => {

    beforeAll(() => {
      render(<ConfirmationDialog {...BASE_PROPS} />);
      component = screen.getByTestId('confirmation-dialog');
    });
  
    it('matches snapshot', () => {
      expect(component).toMatchSnapshot();
    });
  
    it('renders expected title', () => {
      const titleElem = component.getElementsByClassName('ant-result-title');
  
      expect(titleElem[0].innerHTML).toBe('A nice title');
    });
  
    it('renders expected subtitle', () => {
      const subtitleElem = component.getElementsByClassName('ant-result-subtitle');
  
      expect(subtitleElem[0].innerHTML).toBe('A nicer subtitle');
    });
  
    afterAll(() => {
      component.remove();
      component = null;
    });

  describe('handles button events', () => {

    it('calls onClick as expected', () => {
      render(<ConfirmationDialog { ...BASE_PROPS } />);
      const okButton = screen.getByTestId('confirmation-ok-button');
      userEvent.click(okButton);

      expect(onOkClickSpy).toHaveBeenCalledTimes(1);
      okButton.remove();
    });

    it('calls onCancel as expected', () => {
      render(<ConfirmationDialog { ...BASE_PROPS } />);
      const cancelButton = screen.getByTestId('confirmation-cancel-button');
      userEvent.click(cancelButton);

      expect(onCancelClickSpy).toHaveBeenCalledTimes(1);
      cancelButton.remove();
    });

  });

});