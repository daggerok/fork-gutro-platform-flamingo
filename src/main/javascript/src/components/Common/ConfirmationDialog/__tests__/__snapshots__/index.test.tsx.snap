// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmationDialog /> matches snapshot 1`] = `
<div
  class="ant-modal-root"
  data-testid="confirmation-dialog"
>
  <div
    class="ant-modal-mask"
  />
  <div
    class="ant-modal-wrap"
    tabindex="-1"
  >
    <div
      aria-modal="true"
      class="ant-modal"
      role="dialog"
      style="width: 520px;"
    >
      <div
        aria-hidden="true"
        style="width: 0px; height: 0px; overflow: hidden; outline: none;"
        tabindex="0"
      />
      <div
        class="ant-modal-content"
      >
        <div
          class="ant-modal-body"
        >
          <div
            class="ant-result ant-result-warning"
          >
            <div
              class="ant-result-icon"
            >
              <span
                aria-label="warning"
                class="anticon anticon-warning"
                role="img"
              >
                <svg
                  aria-hidden="true"
                  data-icon="warning"
                  fill="currentColor"
                  focusable="false"
                  height="1em"
                  viewBox="64 64 896 896"
                  width="1em"
                >
                  <path
                    d="M955.7 856l-416-720c-6.2-10.7-16.9-16-27.7-16s-21.6 5.3-27.7 16l-416 720C56 877.4 71.4 904 96 904h832c24.6 0 40-26.6 27.7-48zM480 416c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v184c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V416zm32 352a48.01 48.01 0 010-96 48.01 48.01 0 010 96z"
                  />
                </svg>
              </span>
            </div>
            <div
              class="ant-result-title"
            >
              A nice title
            </div>
            <div
              class="ant-result-subtitle"
            >
              A nicer subtitle
            </div>
            <div
              class="ant-result-extra"
            >
              <button
                class="ant-btn ant-btn-default"
                data-testid="confirmation-cancel-button"
                type="button"
              >
                <span>
                  Cancel
                </span>
              </button>
              <button
                class="ant-btn ant-btn-primary"
                data-testid="confirmation-ok-button"
                type="button"
              >
                <span>
                  OK
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <div
        aria-hidden="true"
        style="width: 0px; height: 0px; overflow: hidden; outline: none;"
        tabindex="0"
      />
    </div>
  </div>
</div>
`;
